### PR TITLE
Node Annotation Update:  CoreNodes.FileSystem 

### PR DESCRIPTION
**Pull request by martinstacey**
### Purpose
The following PR addresses nodes' names and descriptions to make them more descriptive for beginners, PRs are organized according to location in DynamoDS.

These corrections are for CoreNodes.FileSystem

**NODE 234 : Image.ReadFromFile**		
Description Input:  File object to load image from.       ->        File object to load image from					
Description Output:  Image      ->        Image object from file									

**NODE 235 : Image.Pixels**							
Description Input:  The image to read.      ->        Image object to get pixel colors from	
Description Output:  Colors at the specified grid points.      ->        Colors at the specified grid points				

**NODE 236 : FileSystem.MoveDirectory**											
Description Input:  string      ->        String representation of existing path						
Description Input:  string      ->        String representation of new path							
Description Input:  bool      ->        Toggle to overwrite existing files				
Description Output:  void      ->        Node performs a task, doesn’t produce an output 									

**NODE 237 : FileSystem.GetDirectoryContents**									
Description:  FileSystem.GetDirectoryContents (directory: var, searchString: string = "*")      ->        Will return a list of files and directories that are contained within a given directory. An optional searchString can be used to filter the results.																
Description Input:  var      ->        Directory to get contents of						
Description Input:  string      ->        Search string used to filter results.
Description Input:  bool      ->        Set to true to include files and folders in subdirectories (recursive) or set to false to include results from top-level of given directory only. 								
Description Output:  var      ->        Resulting files from query		
Description Output:  var      ->        Resulting directories from query
								
**NODE 240 : FileSystem.DeleteFile**															
Description Input:        ->        File path to delete						
Description Output:  void      ->        Node performs a task, doesn’t produce an output 									

**NODE 241 : FileSystem.DeleteDirectory**									
Description:  Deletes a directory.      ->        Deletes a directory							
Description Input:  Path to a directory on disk.      ->        Path to a directory on disk						
Description Output:  void      ->        Node performs a task, doesn’t produce an output 
							
**NODE 242 : FileSystem.CopyFile**									
Description:  Copies a file.      ->        Copies a file						
Description Input:  var      ->        File object to copy							
Description Input:  string      ->        String representation of destination path							
Description Input:  bool      ->        Toggle to overwrite existing files				
Description Output:  void      ->        Node performs a task, doesn’t produce an output 									

**NODE 243 : FileSystem.CopyDirectory**									
Description:  Copies a directory to a destination location.      ->        Copies a directory to a destination location
Description Input:  Directory to copy.      ->        Directory to copy						
Description Input:  Destination of the copy operation on disk.      ->        Destination of the copy operation on disk
Description Input:  bool      ->        Toggle to overwrite existing directory					
Description Output:  void      ->        Node performs a task, doesn’t produce an output 									

**NODE 244 : FileSystem.AppendText**				
Description Output:  void      ->        Node performs a task, doesn’t produce an output 									

**NODE 377 : Image.FromPixels**							
Description Input:  2d rectangular list of colors representing the pixels.      ->        2d rectangular list of colors representing the pixels
Description Output:        ->        Image from 2d list of pixels

**NODE 378 : Image.FromPixels**									
Description Input:  List of colors representing the pixels.      ->        List of colors representing the pixels
Description Output:        ->        Image from list of pixels

**NODE 379 : FileSystem.FileHasExtension**
Description Input:  Path to check for an extension.      ->        Path to check for an extension				
Description Output:        ->        True if file path contains extension, false if it doesn't									

**NODE 380 : FileSystem.FileExtension**						
Description Input:  Path to get extension of.      ->        Path to get extension of							
Description Output:        ->        Extension of file

**NODE 381 : FileSystem.FileExists**					
Description Input:  string      ->        String representing a file path			
Description Output:  bool      ->        True if file exists, false if it doesn't									

**NODE 382 : FileSystem.DirectoryName**									
Description Input:  Path to get directory information of.      ->        Path to get directory information of	
Description Output:  string      ->        Directory name of file path									

**NODE 383 : FileSystem.DirectoryExists**					
Description Input:  Path  to a directory on disk.      ->        Path  to a directory on disk							
Description Output:  bool      ->        True if directory exists, false if it doesn’t								

**NODE 384 : FileSystem.CombinePath**				
Description Input:  String to combine into a path.      ->        String to combine into a path						
Description Output:  string      ->        Combined file path									

**NODE 385 : FileSystem.ChangePathExtension**					
Description Input:  Path to change extension of.      ->        Path to change extension of					
Description Input:  New extension.      ->        String representation of new extension						
Description Output:  string      ->        File path with changed extension

**NODE 548 : FileSystem.FileName**						
Description Input:  Path to get the file name of.      ->        Path to get the file name of					
Description Input:  Determines whether or not the extension is included in the result, defaluts to true.      ->        Toggle to include extension in result		
Description Output:        ->        File name from file path

image

### Reviewers
@Amoursol @aparajit-pratap @QilongTang

### CODE REVIEW

**Comment by Amoursol** [comment]
LGTM @martinstacey !

File: src/Libraries/CoreNodes/FileSystem.cs [hunk 1]
```diff
@@ -76,6 +76,7 @@ public static string ReadText(FileInfo file)
         /// <param name="path"></param>
         /// <param name="newPath"></param>
         /// <param name="overwrite"></param>
+        /// <returns name="void">Node performs a task, doesn’t produce an output </returns>
```
Comment (mmisol) [inline src/Libraries/CoreNodes/FileSystem.cs:79]: I think it's strange to add a `returns` tag when a method does not return anything. Does it actually do anything?

Also, I don't see descriptions for the parameters, which is actually important.

File: src/Libraries/CoreNodes/FileSystem.cs [hunk 2]
```diff
@@ -86,7 +87,8 @@ public static void MoveFile(string path, string newPath, bool overwrite = false)
         /// <summary>
         ///   Deletes the specified file.
         /// </summary>
-        /// <param name="path"></param>
+        /// <param name="path">File path to delete</param>
+        /// <returns name="void">Node performs a task, doesn’t produce an output </returns>
```
Comment (mmisol) [inline src/Libraries/CoreNodes/FileSystem.cs:91]: Same comment about about void methods and `returns` tag.

File: src/Libraries/CoreNodes/FileSystem.cs [hunk 3]
```diff
@@ -95,9 +97,10 @@ public static void DeleteFile(string path)
         /// <summary>
         ///     Copies a file.
         /// </summary>
-        /// <param name="file"></param>
-        /// <param name="destinationPath"></param>
-        /// <param name="overwrite"></param>
+        /// <param name="file">File object to copy</param>
+        /// <param name="destinationPath">String representation of destination path</param>
+        /// <param name="overwrite">Toggle to overwrite existing files</param>
+        /// <returns name="void">Node performs a task, doesn’t produce an output </returns>
```
Comment (mmisol) [inline src/Libraries/CoreNodes/FileSystem.cs:103]: Same comment

File: src/Libraries/CoreNodes/FileSystem.cs [hunk 4]
```diff
@@ -131,6 +135,7 @@ public static void WriteText(string filePath, string text)
         /// </summary>
         /// <param name="filePath">Path to write to</param>
         /// <param name="text">Text content</param>
+        /// <returns name="void">Node performs a task, doesn’t produce an output </returns>
```
Comment (mmisol) [inline src/Libraries/CoreNodes/FileSystem.cs:138]: Same comment

File: src/Libraries/CoreNodes/FileSystem.cs [hunk 5]
```diff
@@ -141,7 +146,8 @@ public static void AppendText(string filePath, string text)
         /// <summary>
         ///     Combines multiple strings into a single file path.
         /// </summary>
-        /// <param name="paths">String to combine into a path.</param>
+        /// <param name="paths">String to combine into a path</param>
```
Comment (mmisol) [inline src/Libraries/CoreNodes/FileSystem.cs:149]: This should say `Strings` as the more common use case is multiple string values.

File: src/Libraries/CoreNodes/FileSystem.cs [hunk 6]
```diff
@@ -219,9 +232,10 @@ public static bool FileHasExtension(string path)
         /// <summary>
         ///     Copies a directory to a destination location.
         /// </summary>
-        /// <param name="directory">Directory to copy.</param>
-        /// <param name="destinationPath">Destination of the copy operation on disk.</param>
-        /// <param name="overwriteFiles"></param>
+        /// <param name="directory">Directory to copy</param>
+        /// <param name="destinationPath">Destination of the copy operation on disk</param>
+        /// <param name="overwriteFiles">Toggle to overwrite existing directory</param>
+        /// <returns name="void">Node performs a task, doesn’t produce an output </returns>
```
Comment (mmisol) [inline src/Libraries/CoreNodes/FileSystem.cs:238]: Same comment regarding methods that don't return anything

File: src/Libraries/CoreNodes/FileSystem.cs [hunk 7]
```diff
@@ -243,8 +257,9 @@ public static void CopyDirectory(DirectoryInfo directory, string destinationPath
         /// <summary>
         ///     Deletes a directory.
         /// </summary>
-        /// <param name="path">Path to a directory on disk.</param>
+        /// <param name="path">Path to a directory on disk</param>
         /// <param name="recursive">Whether or not to delete all contents of the directory, defaults to false.</param>
+        /// <returns name="void">Node performs a task, doesn’t produce an output </returns>
```
Comment (mmisol) [inline src/Libraries/CoreNodes/FileSystem.cs:262]: Same comment

File: src/Libraries/CoreNodes/FileSystem.cs [hunk 8]
```diff
@@ -270,9 +286,10 @@ public static DirectoryInfo DirectoryFromPath(string path)
         /// <summary>
         ///     Moves a directory to a new location.
         /// </summary>
-        /// <param name="path"></param>
-        /// <param name="newPath"></param>
-        /// <param name="overwriteFiles"></param>
+        /// <param name="path">String representation of existing path</param>
+        /// <param name="newPath">String representation of new path</param>
+        /// <param name="overwriteFiles">Toggle to overwrite existing files</param>
+        /// <returns name="void">Node performs a task, doesn’t produce an output </returns>
```
Comment (mmisol) [inline src/Libraries/CoreNodes/FileSystem.cs:292]: Same comment

**Review by mmisol — COMMENTED** [review]
@martinstacey Added some comments. Please address when possible. Thanks!

File: src/Libraries/CoreNodes/FileSystem.cs (hunk 1 above)
Comment (martinstacey) [reply src/Libraries/CoreNodes/FileSystem.cs:79]: this was added to give an explanation into what void means
image

File: src/Libraries/CoreNodes/FileSystem.cs (hunk 5 above)
Comment (martinstacey) [reply src/Libraries/CoreNodes/FileSystem.cs]: Adressed in b7a80b0

File: src/Libraries/CoreNodes/FileSystem.cs (hunk 1 above)
Comment (martinstacey) [reply src/Libraries/CoreNodes/FileSystem.cs:79]: Hi @mmisol, I think I adressed all of the comments on existing Prs, the only thing left to resolve is this

> this was added to give an explanation into what void means
> image

Comment (mmisol) [reply src/Libraries/CoreNodes/FileSystem.cs:79]: Thanks @martinstacey . I was in doubt this had any effect but it seems we are actually using it. Please ignore the first part of my comment.

Please add descriptions for `path`, `newPath` and `overwrite` right above.

File: src/Libraries/CoreNodes/FileSystem.cs [hunk 9]
```diff
@@ -141,16 +146,18 @@ public static void AppendText(string filePath, string text)
         /// <summary>
         ///     Combines multiple strings into a single file path.
         /// </summary>
-        /// <param name="paths">String to combine into a path.</param>
-        public static string CombinePath(params string[] paths)
+        /// <param name="strings">String to combine into a path</param>
```
Comment (mmisol) [inline src/Libraries/CoreNodes/FileSystem.cs:149]: Please make this `Strings to combine into a path` (pluralize `String`)

**Review by mmisol — COMMENTED** [review]
Thanks @martinstacey . Let's keep the comments for `void` functions as they are. Other than that there are two minor things left to fix.

File: src/Libraries/CoreNodes/FileSystem.cs (hunk 1 above)
Comment (martinstacey) [reply src/Libraries/CoreNodes/FileSystem.cs:79]: Addressed in 5e21593

Comment (martinstacey) [reply src/Libraries/CoreNodes/FileSystem.cs:79]: > Thanks @martinstacey . I was in doubt this had any effect but it seems we are actually using it. Please ignore the first part of my comment.
> 
> Please add descriptions for `path`, `newPath` and `overwrite` right above.

Addressed in 5e21593

File: src/Libraries/CoreNodes/FileSystem.cs (hunk 9 above)
Comment (martinstacey) [reply src/Libraries/CoreNodes/FileSystem.cs]: Addressed in a22e753

Comment (martinstacey) [reply src/Libraries/CoreNodes/FileSystem.cs]: @mmisol ok the two things left to fix were addressed.

**Review by mmisol — APPROVED** [review]
Thanks @martinstacey . This one looks good too

**Pull request merged**